### PR TITLE
Make TrajectoryPostProcessor take testable pointer in postprocess()

### PR DESCRIPTION
**Pull request by sniyaz**
By request of @gilwoolee, we want `TrajectoryPostProcessor` derivative classes to take a testable pointer in their `postprocess` methods. This was a feature request relating to the new robot class and the `ParabolicSmoother` class.

In ParabolicTimer, the new field has a default value of `nullptr` and is just ignored.

***

**Before creating a pull request**

- [x] Document new methods and classes
- [x] Format code with `make format`

**Before merging a pull request**

- [x] Set version target by selecting a milestone on the right side
- [x] Summarize this change in `CHANGELOG.md`
- [x] Add unit test(s) for this change

### CODE REVIEW

File: include/aikido/planner/TrajectoryPostProcessor.hpp [hunk 1]
```diff
@@ -13,15 +14,20 @@ class TrajectoryPostProcessor
 public:
   /// \param _inputTraj The untimed trajectory for the arm to process.
   /// \param _rng Random number generator.
+  /// \param _constraint Must be satisfied after prcoessing.
```
Comment (brianhou) [inline include/aikido/planner/TrajectoryPostProcessor.hpp]: Typo: prcoessing -> processing

File: include/aikido/planner/TrajectoryPostProcessor.hpp [hunk 2]
```diff
@@ -13,15 +14,20 @@ class TrajectoryPostProcessor
...
   virtual std::unique_ptr<aikido::trajectory::Spline> postprocess(
       const aikido::trajectory::Interpolated& _inputTraj,
-      const aikido::common::RNG& _rng)
+      const aikido::common::RNG& _rng,
+      const aikido::constraint::TestablePtr& _constraint)
       = 0;
 
   /// \param _inputTraj The untimed *spline* trajectory for the arm to process.
   /// \param _rng Random number generator.
+  /// \param _constraint Must be satisfied after prcoessing.
```
Comment (brianhou) [inline include/aikido/planner/TrajectoryPostProcessor.hpp]: Typo: prcoessing -> processing

**Comment by sniyaz** [comment]
Should be good to go now.

File: CHANGELOG.md [hunk 3]
```diff
@@ -20,6 +20,7 @@
 
 * Planner
 
+  * Make TrajectoryPostProcessor take testable pointer in postprocess(): [#341](https://github.com/personalrobotics/aikido/pull/341)
```
Comment (jslee02) [inline CHANGELOG.md]: Nit: We've used past tense. Could you change this to be in the same fashion?

**Pull request merged**